apm-agent-api/src/main/java/co/elastic/apm/api/Span.java

```diff
@@ -61,39 +61,32 @@
     Span setType(String type);
 
     /**
-     * A flat mapping of user-defined labels with string values.
-     * <p>
-     * Note: in version 6.x, labels are stored under {@code context.tags} in Elasticsearch.
-     * As of version 7.x, they are stored as {@code labels} to comply with ECS (https://github.com/elastic/ecs).
-     * </p>
-     * <p>
-     * Note: the labels are indexed in Elasticsearch so that they are searchable and aggregatable.
-     * By all means,
-     * you should avoid that user specified data,
-     * like URL parameters,
-     * is used as a label key as it can lead to mapping explosions.
-     * </p>
-     *
-     * @param key   The label key.
-     * @param value The label value.
      * @deprecated use {@link #addLabel(String, String)} instead
      */
     @Nonnull
     @Deprecated
     Span addTag(String key, String value);
 
     /**
-     * A flat mapping of user-defined labels with string values.
      * <p>
-     * Note: in version 6.x, labels are stored under {@code context.tags} in Elasticsearch.
-     * As of version 7.x, they are stored as {@code labels} to comply with ECS (https://github.com/elastic/ecs).
+     * Labels are used to add indexed information to transactions, spans, and errors.
+     * Indexed means the data is searchable and aggregatable in Elasticsearch.
+     * Multiple labels can be defined with different key-value pairs.
      * </p>
+     * <ul>
+     *     <li>Indexed: Yes</li>
+     *     <li>Elasticsearch type: <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/object.html">object</a></li>
+     *     <li>Elasticsearch field: {@code labels} (previously {@code context.tags} in stack version {@code < 7.0})</li>
+     * </ul>
      * <p>
-     * Note: the labels are indexed in Elasticsearch so that they are searchable and aggregatable.
-     * By all means,
-     * you should avoid that user specified data,
-     * like URL parameters,
-     * is used as a label key as it can lead to mapping explosions.
+     * Label values can be a string, boolean, or number.
+     * Because labels are stored in the same place in Elasticsearch, each value must have the same data type.
```

Comment:
```suggestion
     * Because labels for a given key are stored in the same place in Elasticsearch, all label values of a given key must have the same data type.
```

Reply:
I mostly copy/pasted that from here: https://github.com/elastic/apm-server/pull/2129
Should we propose the change there as well?

Reply:
I don't know.. What do you think? 
I suggested this change because it wasn't clear enough for me. I actually don't know what 
> stored in the same place in Elasticsearch

really means...

Reply:
I think your changes made it a lot clearer! Would you mind proposing the same changes on the apm-server PR?